src/entry_node.cpp

```diff
@@ -29,16 +29,19 @@
 namespace ovms {
 
 Status EntryNode::execute(session_key_t sessionId, PipelineEventQueue& notifyEndQueue) {
-    // TODO this should be created in EntryNode::SetInputs, or special method for entry node called
-    // in event loop can be done at the end of part 3 or in future release
+    // this should be created in EntryNode::SetInputs, or special method for entry node called
+    // in event loop can be done in future release while implementing dynamic demultiplexing at
+    // entry node
     NodeSessionMetadata metadata;
-    NodeSession& nodeSession = getNodeSession(metadata);  // call to create session
-    notifyEndQueue.push(NodeSessionKeyPair(*this, nodeSession.getSessionKey()));
+    auto nodeSession = getNodeSession(metadata);  // call to create session
+    if (!nodeSession) {
+        return StatusCode::INTERNAL_ERROR;
```

Comment:
Don't we need to push to queue even if execution fails? Otherwise we have no report that entry has finished its execution even in case of errors.